config/config.go

```diff
@@ -956,231 +921,276 @@ func PrintConfig() error {
+	v.SetDefault("Server.TLSKey", filepath.Join(configDir, "certificates", "tls.key"))
+	v.SetDefault("Server.TLSCAKey", filepath.Join(configDir, "certificates", "tlsca.key"))
+	v.SetDefault("Server.SessionSecretFile", filepath.Join(configDir, "session-secret"))
+	v.SetDefault("Xrootd.RobotsTxtFile", filepath.Join(configDir, "robots.txt"))
+	v.SetDefault("Xrootd.ScitokensConfig", filepath.Join(configDir, "xrootd", "scitokens.cfg"))
+	v.SetDefault("Xrootd.Authfile", filepath.Join(configDir, "xrootd", "authfile"))
+	v.SetDefault("Xrootd.MacaroonsKeyFile", filepath.Join(configDir, "macaroons-secret"))
+	v.SetDefault("IssuerKey", filepath.Join(configDir, "issuer.jwk"))
+	v.SetDefault("Server.UIPasswordFile", filepath.Join(configDir, "server-web-passwd"))
+	v.SetDefault("Server.UIActivationCodeFile", filepath.Join(configDir, "server-web-activation-code"))
+	v.SetDefault("OIDC.ClientIDFile", filepath.Join(configDir, "oidc-client-id"))
+	v.SetDefault("OIDC.ClientSecretFile", filepath.Join(configDir, "oidc-client-secret"))
+	v.SetDefault("Server.WebConfigFile", filepath.Join(configDir, "web-config.yaml"))
+	v.SetDefault("Cache.ExportLocation", "/")
+	v.SetDefault("Registry.RequireKeyChaining", true)
+	v.SetDefault("Origin.StorageType", "posix")
+	v.SetDefault("Origin.SelfTest", true)
+	v.SetDefault("Origin.DirectorTest", true)
 	// Set up the default S3 URL style to be path-style here as opposed to in the defaults.yaml becase
 	// we want to be able to check if this is user-provided (which we can't do for defaults.yaml)
-	viper.SetDefault("Origin.S3UrlStyle", "path")
-
-	if param.Cache_DataLocation.IsSet() {
-		log.Warningf("Deprecated configuration key %s is set. Please migrate to use %s instead", param.Cache_DataLocation.GetName(), param.Cache_LocalRoot.GetName())
-		log.Warningf("Will attempt to use the value of %s as default for %s", param.Cache_DataLocation.GetName(), param.Cache_LocalRoot.GetName())
-	}
+	v.SetDefault("Origin.S3UrlStyle", "path")
 
 	if IsRootExecution() {
-		if currentServers.IsEnabled(server_structs.OriginType) {
-			viper.SetDefault("Origin.RunLocation", filepath.Join("/run", "pelican", "xrootd", "origin"))
-		}
-		if currentServers.IsEnabled(server_structs.CacheType) {
-			viper.SetDefault("Cache.RunLocation", filepath.Join("/run", "pelican", "xrootd", "cache"))
-		}
+		v.SetDefault("Origin.RunLocation", filepath.Join("/run", "pelican", "xrootd", "origin"))
+		v.SetDefault("Cache.RunLocation", filepath.Join("/run", "pelican", "xrootd", "cache"))
 
 		// To ensure Cache.DataLocation still works, we default Cache.LocalRoot to Cache.DataLocation
 		// The logic is extracted from handleDeprecatedConfig as we manually set the default value here
-		viper.SetDefault(param.Cache_DataLocation.GetName(), "/run/pelican/cache")
-		viper.SetDefault(param.Cache_LocalRoot.GetName(), param.Cache_DataLocation.GetString())
+		v.SetDefault("Cache.DataLocation", "/run/pelican/cache")
```

Comment:
I think there's a small bug somewhere in the way this variable is treated. Trying to serve a local origin where I can guarantee I've not touched the `Cache.DataLocation` parameter results in this snippet of printed config:
```
{
  "Cache": {
    "Concurrency": 0,
    "DataLocation": "",
    "DataLocations": null,
```

Yet I also get this logged warning:
```
WARNING[2024-11-04T17:01:58Z] Deprecated configuration key Cache.DataLocation is set. Please migrate to use Cache.LocalRoot instead 
WARNING[2024-11-04T17:01:58Z] Will attempt to use the value of Cache.DataLocation as default for Cache.LocalRoot 
```

Reply:
Addressed it in the latest series of commits.